rest/route/project.go

```diff
@@ -656,6 +656,86 @@ func (h *projectRepotrackerHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// DELETE /rest/v2/projects/{project_id}
+
+type projectDeleteHandler struct {
+	projectName string
+	sc          data.Connector
+}
+
+func makeDeleteProject(sc data.Connector) gimlet.RouteHandler {
+	return &projectDeleteHandler{
+		sc: sc,
+	}
+}
+
+func (h *projectDeleteHandler) Factory() gimlet.RouteHandler {
+	return &projectDeleteHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *projectDeleteHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.projectName = gimlet.GetVars(r)["project_id"]
+	return nil
+}
+
+func (h *projectDeleteHandler) Run(ctx context.Context) gimlet.Responder {
+	project, err := dbModel.FindOneProjectRef(h.projectName)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(err)
+	}
+	if project == nil {
+		return gimlet.MakeJSONErrorResponder(errors.Errorf("project '%s' does not exist", h.projectName))
+	}
+
+	if project.Hidden {
+		return gimlet.MakeJSONErrorResponder(errors.Errorf("project '%s' is already hidden", h.projectName))
+	}
+
+	if !project.UseRepoSettings {
+		return gimlet.MakeJSONErrorResponder(
+			errors.Errorf("project '%s' must have UseRepoSettings enabled to be eligible for deletion", h.projectName))
+	}
+
+	skeletonProj := dbModel.ProjectRef{
+		Id:              project.Id,
+		Owner:           project.Owner,
+		Repo:            project.Repo,
+		Branch:          project.Branch,
+		RepoRefId:       project.RepoRefId,
+		Enabled:         false,
+		UseRepoSettings: true,
+		Hidden:          true,
+	}
+	if err = skeletonProj.Update(); err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "project '%s' could not be updated", project.Id))
```

Comment:
One other thing is that for DB errors (errors between 714 to 733) you should use MakeJSONInternalErrorResponder, so that the error code given to the user is InternalServerError and not BadRequest. (I don't need to see this for you to merge though)